include/audit_events.hpp

```diff
@@ -0,0 +1,295 @@
+{
+    return (req.target() == "/redfish/v1/SessionService/Sessions") ||
+           (req.target() == "/redfish/v1/SessionService/Sessions/") ||
+           (req.target() == "/login");
+}
+
+/**
+ * @brief Checks if request should be audited after completion
+ * @return  True if request should be audited
+ */
+inline bool wantAudit(const crow::Request& req)
+{
+    return (req.method() == boost::beast::http::verb::patch) ||
+           (req.method() == boost::beast::http::verb::put) ||
+           (req.method() == boost::beast::http::verb::delete_) ||
+           ((req.method() == boost::beast::http::verb::post) &&
+            !checkPostUser(req));
+}
+
+/**
+ * @brief Checks if request should include additional data
+ *
+ * - Accounts requests data may contain passwords
+ * - IBM Management console events data is not useful. It can be binary data or
+ *   contents of file.
+ * - User login and session data may contain passwords
+ *
+ * @return True if request's data should not be logged
+ */
+inline bool checkSkipDetail(const crow::Request& req)
+{
+    return req.target().starts_with("/redfish/v1/AccountService/Accounts") ||
+           req.target().starts_with("/ibm/v1") ||
+           ((req.method() == boost::beast::http::verb::post) &&
+            checkPostUser(req));
+}
+
+/**
+ * @brief Checks if request's detail data should be logged
+ *
+ * @return True if request's detail data should be logged
+ */
+inline bool wantDetail(const crow::Request& req)
+{
+    switch (req.method())
+    {
+        case boost::beast::http::verb::patch:
+        case boost::beast::http::verb::post:
+            if (checkSkipDetail(req))
+            {
+                return false;
+            }
+            return true;
+
+        case boost::beast::http::verb::put:
+            return (!req.target().starts_with("/ibm/v1"));
+
+        case boost::beast::http::verb::delete_:
+            return true;
+
+        default:
+            // Shouldn't be here, don't log any data
+            BMCWEB_LOG_DEBUG("Unexpected verb {}", req.methodString());
+            return false;
+    }
+}
+
+/**
+ * @brief Appends item to strBuf only if strBuf won't exceed maxBufSize
+ *
+ * @param[in,out] strBuf Buffer to append up to maxBufSize only
+ * @param[in] maxBufSize Maximum length of strBuf
+ * @param[in] item String to append if it will fit within maxBufSize
+ *
+ * @return True if item was appended
+ */
+inline bool appendItemToBuf(std::string& strBuf, size_t maxBufSize,
+                            const std::string& item)
+{
+    if (strBuf.length() + item.length() > maxBufSize)
+    {
+        return false;
+    }
+    strBuf += item;
+    return true;
+}
+
+inline void auditEvent(const crow::Request& req, const std::string& userName,
+                       bool success)
+{
+    if (!auditOpen())
+    {
+        return;
+    }
+
+    std::string opPath = "op=" + std::string(req.methodString()) + ":" +
+                         std::string(req.target()) + " ";
```

Comment:
Perhaps, it may be easier to read if we do 
```
std::string opPath = std::format("op={}:{} ",  req.methodString(),  req.target());
```

Reply:
Thank you, I agree that is lots easier to read. (In fact it is also one of the comments from upstream as well.) I will make this change.